src/clj/medusa/db.clj

```diff
@@ -242,7 +242,7 @@
                      (map :user_id)
                      (apply hash-set))
 
-        subs (let [subs (set/union metric-subs detector-subs)]
```

Comment:
After removing `detector-subs` here, I believe `set/union` is no longer needed, and indeed the `let` itself becomes redundant since you can map on metric-subs directly. This also removes the need to define `detector-subs` above it in this function. With these changes combined, I believe the function could possibly simplified to following (untested):

```clj
(defn get-subscribers-for-metric [id]
  (let [metric-subs (->>
                     (select metric
                             (fields :id)
                             (where (= :id id))
                             (with user
                                   (fields [:id :user_id])))
                     first
                     :user
                     (map :user_id)
                     (apply hash-set))

        subs (map #(-> (select user (where (= :id %)))
                         first
                         :email)
                    metric-subs)]
    subs))
```

Reply:
Nope, no worries. I have zero clj experience and am unsure how to test my changes beyond pushing to production.

I think I will just keep it the way it is for now, as even your refactored approach includes some things that are largely unused (individuals cannot any longer subscribe to metrics since we removed Persona login. We rely entirely on the alert_emails fields now). But once we start down that rabbit hole I really hope we have a better plan for testing and deploying changes.

Reply:
Understood; let me know if you'd like to have a higher bandwidth discussion later about testing, deployment, etc for this project.

Reply:
I'd love to hear your thoughts on Clojure testing frameworks. The first thing I like to do when encounting a new, testless project is write a couple of tests to keep me from coding off a cliff. :-)

Reply:
I'd love to tell you them, but I don't have any thoughts on Clojure testing frameworks :S 

So I'm happy to hear yours!

Reply:
The built-in clojure.test seems like a good start: https://clojure.github.io/clojure/clojure.test-api.html.